src/hardware/setup.rs

```diff
@@ -1003,6 +1009,67 @@ pub fn setup(
         lvds7: gpiod.pd4.into_push_pull_output(),
     };
 
+    let (usb_device, usb_serial) = {
+        let usb_bus = cortex_m::singleton!(: Option<usb_device::bus::UsbBusAllocator<UsbBus>> = None).unwrap();
+        let endpoint_memory =
+            cortex_m::singleton!(: [u32; 1024] = [0; 1024]).unwrap();
+
+        //let usb_id = gpioa.pa10.into_alternate::<8>();
+        let usb_n = gpioa.pa11.into_alternate();
+        let usb_p = gpioa.pa12.into_alternate();
+
+        let usb = stm32h7xx_hal::usb_hs::USB2::new(
+            device.OTG2_HS_GLOBAL,
+            device.OTG2_HS_DEVICE,
+            device.OTG2_HS_PWRCLK,
+            usb_n,
+            usb_p,
+            ccdr.peripheral.USB2OTG,
+            &ccdr.clocks,
+        );
+
+        // Generate a device serial number from the MAC address.
+        let serial_number =
+            cortex_m::singleton!(: Option<heapless::String<64>> = None)
```

Comment:
17 is enough.

Reply:
Fixed in 2bd057c35fc9230293f612668f01fc5aff581371